src/mappings/Rewards.ts

```diff
@@ -404,7 +404,9 @@ async function buildRewardEvents<A>(
     [initialInnerAccumulator, []],
   );
 
-  await Promise.allSettled(savingPromises);
+  for (const promise of savingPromises) {
+    await promise;
+  }
```

Comment:
Creating the promises is still happening in parallel, I think the reduce will need to be changed